### DISCUSSION TITLE
Träwelling Year in Review 2024

### DISCUSSION THREAD

**Discussion by MrKrisKrisu** [Traewelling/traewelling#3014] [Announcements]
_[english version below]_
Liebe Community,

in den letzten drei Jahren haben wir euch mit unserem „Year in Review” eine Möglichkeit geboten, eure persönlichen Statistiken des Jahres anzusehen. Dieses Jahr fehlt uns leider die Zeit und Energie, um eine entsprechende Seite bereitzustellen.

Dennoch möchten wir euch die Möglichkeit nicht nehmen:

Der API-Endpoint, den wir bisher genutzt haben, ist ab sofort wieder aktiv und kann verwendet werden. Wir haben den Endpoint in [unsere API-Dokumentation](https://traewelling.de/api/documentation#/Statistics/getYearInReview) aufgenommen, wenn auch nicht mit der vollen Response.

Mit einem Aufruf von `https://traewelling.de/api/v1/year-in-review` _(Ihr müsst entweder im Browser eingeloggt sein oder euch per API Token authentifizieren)_ erhaltet ihr eine entsprechende Statistik des aktuellen Jahres. Die Response ist dann für eine Woche gecached. Nach Ablauf dieser Woche wird die Statistik neu generiert, damit auch bis zum Ende des Jahres eine volle Statistik angezeigt werden kann. Der Aufruf des Endpunkts ist recht aufwändig, da viele Daten prozessiert werden und kann daher ein paar Sekunden dauern.

Wir möchten uns dafür entschuldigen, dass wir dieses Jahr keine vollständige Lösung anbieten können. Gleichzeitig freuen wir uns, wenn jemand aus der Community Lust hat, den aktivierten API-Endpoint zu nutzen und vielleicht etwas Eigenes damit zu entwickeln.

Euer Träwelling-Team
Kris, @HerrLevin und @jeyemwey

--------------

Dear Community,

In the last three years, we have offered you the opportunity to view your personal statistics for the year with our „Year in Review”. This year, unfortunately, we don't have the time and energy to provide a corresponding page.

Nevertheless, we don't want to deprive you of this opportunity:

The API endpoint that we used is now active again. We have included the endpoint in [our API documentation](https://traewelling.de/api/documentation#/Statistics/getYearInReview), but not with a full documnetation of the response.

By calling `https://traewelling.de/api/v1/year-in-review` _(you must either be logged into the browser or authenticate yourself via API token)_ you will receive the corresponding statistics for the current year. The response is then cached for one week. At the end of this week, the statistics are regenerated so that full statistics can be displayed until the end of the year. Calling the endpoint is quite time-consuming as a lot of data is processed and can therefore take a few seconds.

We would like to apologize for not being able to offer a complete solution this year. At the same time, we would be delighted if someone from the community would like to use the activated API endpoint and perhaps develop something of their own with it.

Your Träwelling team
Kris, @HerrLevin and @jeyemwey

**Comment by marhei** [comment]
Hallo liebe Träwelling-Community,

danke für den Ansporn von euch und @jheubuch, ich hab mir dann mal die Mühe gemacht und habe für Lokbuch auf iOS ein Jahresrückblick realisiert. Lokbuch ist aktuell über TestFlight mit dem folgenden Link auf jedem Gerät, dass mindestens auf iOS 17 läuft, installierbar: https://testflight.apple.com/join/8djTU1G6

![image](https://github.com/user-attachments/assets/c4feef1a-a635-4ec3-984c-97ca9af9f301)
![image](https://github.com/user-attachments/assets/00d6feb6-59f2-40d8-8aa8-1a8557a86ef7)
![image](https://github.com/user-attachments/assets/2e3fc2d0-4109-40c7-baaf-527a0badca0f)
![image](https://github.com/user-attachments/assets/91555e65-e700-4136-9f2f-a4f03226cdb7)

**Comment by jheubuch** [comment]
Hallo zusammen,

wie @marhei es schon erwähnt hat, habe ich auch für @Traewelldroid einen kleinen Recap gebaut. Das sieht ungefähr so aus:

![1398cfc2-8596-4593-8514-465cf72f05a64048717875719363524.png](https://github.com/user-attachments/assets/78821f9f-4e6c-4878-9aae-a05c8cec108a)

![09dda950-6414-4031-99aa-9d0328c9ffb8108755000809943073.png](https://github.com/user-attachments/assets/ff4a0d29-baa8-4428-8926-59fa327a9d0a)

und einiges mehr.

Teilt gerne euer letztes Fahrtenjahr mit euren Freunden oder in sozialen Netzwerken, ich bin gespannt, zu sehen, wie viel ihr alle so unterwegs wart! ☺️

Vielen Dank an dieser Stelle auch an das @Traewelling team, was die Daten so schön aufbereitet zur Verfügung stellt 💙
